src/ranged.cpp

```diff
@@ -150,8 +150,11 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
              target.x, target.y, target.z );
 
     // Trace the trajectory, doing damage in order
-    tripoint &tp = ret.end_point;
+    tripoint &tp = attack.end_point;
     tripoint prev_point = source;
+    std::vector<tripoint> trajectory_extension = continue_line( trajectory, proj_arg.range );
+    trajectory.resize( trajectory.size() + trajectory_extension.size() );
+    trajectory.insert( trajectory.end(), trajectory_extension.begin(), trajectory_extension.end() );
```

Comment:
Is this correct? The way I understand it, it will insert `trajectory_extension.size()` default value tripoints between the end of original trajectory and the new extended one.
Shouldn't it be `reserve` instead of `resize`?